src/web/hooks/useTiming.js

```diff
@@ -0,0 +1,109 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {useEffect, useCallback, useState} from 'react';
+
+import logger from 'gmp/log';
+
+import {hasValue, isFunction, isDefined} from 'gmp/utils/identity';
+
+import useInstanceVariable from './useInstanceVariable';
+
+const log = logger.getLogger('web.hooks.useTiming');
+
+/**
+ * Hook to start a timer that calls a function after a given timeout
+ *
+ * @param {Function} doFunc The function to call
+ * @param {Number|Function} timeout The timeout in milliseconds or a function that returns the timeout
+ * @returns Array of startTimer function, clearTimer function and boolean isRunning
+ */
+const useTiming = (doFunc, timeout) => {
+  const timer = useInstanceVariable({});
+  const [timerId, setTimerId] = useState(); // store timerId in state too to trigger re-render if it changes
+  const isRunning = !!timerId;
```

Comment:
Nitpick: I would avoid `!!`
```suggestion
  const isRunning = Boolean(timerId);
```

Reply:
Any reason for avoiding `!!`? GSA uses `!!` for boolean conversion at several places at the moment.